src/bitcoin-cli.cpp

```diff
@@ -349,6 +352,72 @@ class NetinfoRequestHandler : public BaseRequestHandler
         const double milliseconds{round(1000 * seconds)};
         return milliseconds > 999999 ? "-" : ToString(milliseconds);
     }
+    std::string ConnectionTypeForNetinfo(const std::string& conn_type) const
+    {
+        if (conn_type == "outbound-full-relay" || conn_type == "inbound") return "full";
+        if (conn_type == "block-relay-only") return "block";
+        if (conn_type == "manual" || conn_type == "feeler") return conn_type;
+        if (conn_type == "addr-fetch") return "addr";
```

Comment:
Yes, if an integral or enum connection type was exposed, but getpeerinfo provides it as a string, so this is the best we can do, I think. 

https://en.cppreference.com/w/cpp/language/switch

Reply:
(Client side, I think we have to make do data-wise with what the server side rpc provides. We can't hook in here to CNodeStats; it would be a layer violation).

Reply:
Yea it's too bad that C++ doesn't have case pattern matching on strings, otherwise that'd be a lot cleaner.
(this applies to some other places as well such as the P2P message dispatch)